PassleSync.Core/ContentFinders/BaseContentFinder.cs

```diff
@@ -27,23 +30,42 @@ public bool TryFindContent(PublishedRequest request)
                 return false;
             }
 
-            if (request.Uri.Segments.Skip(1).Take(1).SingleOrDefault() != $"{UrlPrefix}/")
+            var virtualContentType = request.UmbracoContext.Content.GetContentType(ContentType);
+            var virtualContent = request.UmbracoContext.Content.GetByContentType(virtualContentType);
+            
+            if (virtualContent == null)
             {
                 return false;
             }
 
-            var shortcode = request.Uri.Segments.Reverse().Skip(1).Take(1).SingleOrDefault().Trim('/');
+            string shortcode = string.Empty;
+            if (ContentType == PassleContentType.PASSLE_AUTHOR || ContentType == PassleContentType.PASSLE_POST) 
+            {
+                // Create the regex pattern using the shortcode name and slug name
+                string regexPattern = UrlTemplate
+                       .Replace($"{{{{{ShortcodeName}}}}}", $"(?<{ShortcodeName}>.+)")
+                       .Replace($"{{{{{SlugName}}}}}", $"(?<{SlugName}>.+)");
 
-            var virtualContentType = request.UmbracoContext.Content.GetContentType(ContentType);
-            var virtualContent = request.UmbracoContext.Content.GetByContentType(virtualContentType);
+                // Create a regular expression object and match it against the request URI
+                Regex regex = new Regex(regexPattern);
+                Match match = regex.Match(request.Uri.ToString());
 
-            if (virtualContent == null)
+                if (match.Success)
+                {
+                    // Extract the shortcode value from the named group
+                    shortcode = match.Groups[ShortcodeName].Value;
+                }
+                else
+                {
+                    shortcode = request.Uri.Segments.Reverse().Skip(1).Take(1).SingleOrDefault().Trim('/');
```

Comment:
When will this code be run? How do we know the shortcode will be the last segment in the path?

Reply:
I just kept this as a fallback but maybe we don't need this - I was mostly thinking about people that use prefixes